backend/src/Service/Admin/Order/AdminOrderService.php

```diff
@@ -976,4 +978,29 @@ public function resolveOrderHasPayConflictAnswersByAdmin(OrderHasPayConflictAnsw
 
         return $response;
     }
+
+    // Update isCashPaymentConfirmedByStore by admin
+    public function updateIsCashPaymentConfirmedByStoreByAdmin(OrderUpdateIsCashPaymentConfirmedByStoreByAdminRequest $request, int $userId): ?OrderUpdateIsCashPaymentConfirmedByStoreForAdminResponse
+    {
+        $order = $this->adminOrderManager->updateIsCashPaymentConfirmedByStoreByAdmin($request);
+
+        if ($order) {
+            // create order log in order time line
+            $this->orderTimeLineService->createOrderLogsRequest($order);
+
+            // save log of the action on order
+            $this->orderLogService->createOrderLogMessage($order, $userId, OrderLogCreatedByUserTypeConstant::ADMIN_USER_TYPE_CONST,
+                OrderLogActionTypeConstant::CONFIRM_CAPTAIN_PAID_TO_PROVIDER_BY_STORE_VIA_DASHBOARD_CONST, null, null);
+
+            // send firebase notification to admin if the captain’s answer differs from that of the store, regarding the field (paidToProvider and isCashPaymentConfirmedByStore)
+            if ($order->getIsCashPaymentConfirmedByStore() !== $order->getPaidToProvider()) {
```

Comment:
1. S from SOLID is broken here.
2. may we can use messenger so we can save > 400 ms.

Reply:
We already have used messenger to save order log, and that's happening here: $this->orderLogService->createOrderLogMessage